[*] Running ...
[*] Dataset loaded: D:\MCTS-AHD-master\problems\tsp_constructive\dataset\train50_dataset.npy with 64 instances.
D:\MCTS-AHD-master\problems\tsp_constructive\gpt.py:30: RuntimeWarning: divide by zero encountered in scalar divide
  scaled_detour_penalty = detour_penalty * (1 + (centrality / direct_distance))
D:\MCTS-AHD-master\problems\tsp_constructive\gpt.py:30: RuntimeWarning: invalid value encountered in scalar multiply
  scaled_detour_penalty = detour_penalty * (1 + (centrality / direct_distance))
Traceback (most recent call last):
  File "D:\MCTS-AHD-master\problems\tsp_constructive\eval.py", line 82, in <module>
    obj = eval_heuristic(node_positions[i])
  File "D:\MCTS-AHD-master\problems\tsp_constructive\eval.py", line 52, in eval_heuristic
    raise KeyError(f"Node {next_node} is already visited.")
KeyError: 'Node None is already visited.'
